sphinx_gallery/notebook.py

```diff
@@ -118,6 +126,8 @@ def __init__(self, file_name, target_dir):
         self.write_file = os.path.join(target_dir, self.file_name)
         self.work_notebook = ipy_notebook_skeleton()
         self.add_code_cell("%matplotlib inline")
+        self.fill_notebook(script_blocks)
+        self.save_file()
```

Comment:
It's generally considered best practice to have minimal logic in the constructor. In this case I would rather have an API like:

```py

nb = Notebook(source_filename)
nb.to_notebook(target_filename)
```

The parsing of the blocks from `source_filename` could happen in the constructor.